File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] answers = new int[nums.length];
        int[] rightProducts = new int[nums.length];
        int right = 1;
        int left = 1;
        rightProducts[nums.length-1] = right;
        for( int i = nums.length-2 ; i >= 0 ; i-- ){
            right *= nums[i+1];
            rightProducts[i] = right;
        }
        answers[0] = left*rightProducts[0];
        for( int i = 1 ; i < nums.length; i++ ) {
            left *= nums[i-1];
            answers[i] = left*rightProducts[i];
        }
        return answers;
    }


    
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)-1
        while( left <= right ):
            medium = (right+left)//2
            if nums[medium] == target: 
                return medium
            if nums[medium] < target:
                left = medium+1
            else:
                right = medium-1
        return -1